### ISSUE TITLE
Blind Accessibility

### ISSUE THREAD

**Issue by Jaseoffire** [mapeditor/tiled#3948]
**Is your feature request related to a problem? Please describe.**
This feature request is to mainly add/improve screen reader support. Right now, the start screen is somewhat accessible. With a screen reader, I am able to see the buttons for creating a new project, opening a project, creating a new map, and creating a new tileset.
**Describe the solution you'd like**
The idea is that through QT's own accessibility features, and perhaps through other screen reader support, we can create a mostly usable program for the visually impaired. This should have minimal impact on other features of the GUI.
**Describe alternatives you've considered**
I've considered first just making an entire new setup, but TMX is far too used and valuable. Even building a fully featured TMX editor would be cumbersome. As far as alternatives to the idea above, it is possible, and might be necessary to add an internal voicing system to maintain cross-platform compatibility.

**Comment by azurejoga** [comment]
Supported!

**Comment by mcb2003** [comment]
As a blind game developer, I would love to see this happen!

**Comment by bjorn** [comment]
I've marked this issue as "help wanted", because I have no experience at all with accessibility features and also do not know how a visual editor like Tiled could actually be made accessible for blind people. The existing accessibility of the start screen mentioned by @Jaseoffire is due to Qt's built-in accessibility features based on the standard widget hierarchy. Beyond that I expect nothing to work at the moment since the main feature of the app is the editable view on the map and all edit tools rely almost exclusively on mouse pointer input.

Note that help does not necessarily mean actual coding. This issue can also use help from people who can describe existing solutions to making such an editor accessible and refer to other applications that might already have implemented them.

**Comment by deadowl** [comment]
I'd be down for helping with this. Most of my accessibility testing experience is with web browsers but I know some of the basic concepts of using a screenreader and wouldn't be bothered by learning more. Add onto that, not having a great deal of experience with the tools you're using is beneficial from a usability analysis perspective. Generally speaking, the goal is to be able to use keyboard input for any operation, and there's a lot that can be shored up before even approaching that.

On the web I'm mostly worried about tab order, tab traps, links, headers, descriptive text for images, contrast, and you can probably throw in avoiding photo-sensitive epilepsy issues (e.g. in Tiled you might want to turn off animation rendering by default and also ensure it isn't easy to turn on by accident).

I don't exactly have JAWS which is what most people use, but Windows comes with a built-in screenreader called Narrator, and pretty much all operating systems these days have  screenreader built in. The most immediate top-level problem I notice is that there are a number of entries in the Layer menu that don't get highlighted by Narrator or read by narrator when navigating that menu--doesn't seem to be a problem for File, Edit, View, World, Map, TIleset, Project or Help.

Probably the most annoying thing I've found so far is that the screenreader stops reading things to you when you're in the properties editor and then switch out of it. Says something like "scan off" then I don't know what to do from there. Also not sure how to find visibility into disabled menu items.

**Comment by deadowl** [comment]
For map tooling, providing a dialog to query of what's at x,y for a given layer and applying a tool targeting x,y for a given layer would be immensely helpful as a substitute for relying on mouse actions if there's no present way to do this.

**Comment by retro-coder64** [comment]
In my experience of using a screen reader I found that some of the most successful apps are terminal based. Especially with the MCP’s becoming more popular it could be possible to build an AI assistant that will perform most tasks for you such as placing tiles. Another option could be for the UI to be rewritten to expose itself to Windows accessibility APIs.
